webrtc/RTCPeerConnection-set-description.html

```diff
@@ -0,0 +1,395 @@
+<!doctype html>
+<title>RTCPeerConnection Set Description tests</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script type="text/javascript">
+  'use strict';
+
+  const audioLineRegex = /\r\nm=audio.+\r\n/g
+  const videoLineRegex = /\r\nm=video.+\r\n/g
+
+  function countAudioLine(sdp) {
+    const matches = sdp.match(audioLineRegex);
+    if(matches === null) {
+      return 0;
+    } else {
+      return matches.length;
+    }
+  }
+
+  function countVideoLine(sdp) {
+    const matches = sdp.match(videoLineRegex);
+    if(matches === null) {
+      return 0;
+    } else {
+      return matches.length;
+    }
+  }
+
+  function similarMediaDescriptions(sdp1, sdp2) {
+    if(sdp1 === sdp2) {
+      return true;
+    } else if(countAudioLine(sdp1) !== countAudioLine(sdp2)) {
+      return false;
+    } else if(countVideoLine(sdp1) !== countVideoLine(sdp2)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function assert_defined(object) {
+    if(object === undefined || object === null)
+      assert_unreached('object must not be undefined or null');
+  }
+
+  function isSimilarSdp(sdp1, sdp2) {
+    assert_defined(sdp1);
```

Comment:
I have added an `assert_is_session_description` to test if the object is either instance of `RTCSessionDescription` or plain object. 

Speaking of which I found a minor bug in Chrome which is that their `createOffer` and `createAnswer` returns a `RTCSessionDescription` instead of `RTCSessionDescriptionInit` object as per the spec. Not sure if it is worth to test that in WPT or should simply file a bug to Chromium.